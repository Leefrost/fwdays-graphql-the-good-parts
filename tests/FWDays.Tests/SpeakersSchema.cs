using FWDays.Speakers;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

using Mutations = FWDays.Speakers.Mutations;
using Queries = FWDays.Speakers.Queries;

namespace FWDays.Tests;

public class SpeakersSchema
{
    [Fact]
    public async Task SchemaChanged()
    {
        var builder = new ServiceCollection()
            .AddDbContextPool<SpeakersDbContext>(
                options => options.UseInMemoryDatabase("Data Source=fwdays.db"))
            .AddGraphQL()
            .AddQueryType(d => d.Name("Query"))
            .AddTypeExtension<Queries>()
            .AddMutationType(d => d.Name("Mutation"))
            .AddTypeExtension<Mutations>();
        
        var schema = await builder.BuildSchemaAsync();

        schema.Print().MatchSnapshot();
    }
}