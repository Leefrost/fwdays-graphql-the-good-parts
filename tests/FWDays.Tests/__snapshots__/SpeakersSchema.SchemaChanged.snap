schema {
  query: Query
  mutation: Mutation
}

type Mutation {
  registerSpeaker(input: SpeakerInput!): SpeakerRegistrationPayload!
}

"Information about pagination in a connection."
type PageInfo {
  "Indicates whether more edges exist following the set defined by the clients arguments."
  hasNextPage: Boolean!
  "Indicates whether more edges exist prior the set defined by the clients arguments."
  hasPreviousPage: Boolean!
  "When paginating backwards, the cursor to continue."
  startCursor: String
  "When paginating forwards, the cursor to continue."
  endCursor: String
}

type Query {
  speakers("Returns the first _n_ elements from the list." first: Int "Returns the elements in the list that come after the specified cursor." after: String "Returns the last _n_ elements from the list." last: Int "Returns the elements in the list that come before the specified cursor." before: String): SpeakersConnection
}

type Speaker {
  id: Int!
  firstName: String!
  lastName: String!
  company: String!
  position: String!
  bio: String
  topic: String!
  github: String
}

type SpeakerRegistrationPayload {
  id: Int!
  firstName: String!
  lastName: String!
  topic: String!
}

"A connection to a list of items."
type SpeakersConnection {
  "Information to aid in pagination."
  pageInfo: PageInfo!
  "A list of edges."
  edges: [SpeakersEdge!]
  "A flattened list of the nodes."
  nodes: [Speaker!]
}

"An edge in a connection."
type SpeakersEdge {
  "A cursor for use in pagination."
  cursor: String!
  "The item at the end of the edge."
  node: Speaker!
}

input SpeakerInput {
  firstName: String!
  lastName: String!
  company: String!
  position: String!
  bio: String!
  topic: String!
  github: String
}
